saleor/checkout/migrations/0072_alter_checkoutline_undiscounted_unit_price_amount.py

```diff
@@ -0,0 +1,29 @@
+# Generated by Django 4.2.15 on 2024-10-29 11:02
+
+from decimal import Decimal
+
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("checkout", "0071_propagate_line_unidscounted_unit_price"),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name="checkoutline",
+            name="undiscounted_unit_price_amount",
+            field=models.DecimalField(
+                decimal_places=3, default=Decimal("0"), max_digits=12
+            ),
+        ),
+        migrations.RunSQL(
+            """
+            ALTER TABLE checkout_checkoutline
+            ALTER COLUMN undiscounted_unit_price_amount
+            SET DEFAULT 0;
+            """,
+            migrations.RunSQL.noop,
+        ),
```

Comment:
We should split state and db operation here:
```suggestion
        migrations.SeparateDatabaseAndState(
            state_operations=[
                migrations.AlterField(
                    model_name="checkoutline",
                    name="undiscounted_unit_price_amount",
                    field=models.DecimalField(
                        decimal_places=3, default=Decimal("0"), max_digits=12
                    ),
                ),
            ],
            database_operations=[
                migrations.RunSQL(
                    """
                    ALTER TABLE checkout_checkoutline
                    ALTER COLUMN undiscounted_unit_price_amount
                    SET DEFAULT 0;
                    """,
                    migrations.RunSQL.noop,
                ),
                migrations.RunSQL(
                    """
                    ALTER TABLE checkout_checkoutline
                    ALTER COLUMN undiscounted_unit_price_amount
                    SET NOT NULL;
                    """
                )
            ]
        )
```

This way we will not generate this query in the migrations:
`UPDATE "checkout_checkoutline" SET "undiscounted_unit_price_amount" = 0 WHERE "undiscounted_unit_price_amount" IS NULL; SET CONSTRAINTS ALL IMMEDIATE;`

Reply:
Thanks @kadewu  ❤️